bsp/m5stack_core/m5stack_core.c

```diff
@@ -0,0 +1,476 @@
+        .pin_bit_mask = (1ULL << BSP_SPEAKER_IO),
+        .mode = GPIO_MODE_OUTPUT,
+        .pull_up_en = GPIO_PULLUP_DISABLE,
+        .pull_down_en = GPIO_PULLDOWN_DISABLE,
+        .intr_type = GPIO_INTR_DISABLE,
+    };
+    ESP_ERROR_CHECK(gpio_config(&io_conf));
+
+    gpio_set_level(BSP_SPEAKER_IO, 0);  // Turn off speaker
+
+    return ESP_OK;
+}
+
+// Bit number used to represent command and parameter
+#define LCD_CMD_BITS   8
+#define LCD_PARAM_BITS 8
+#define LCD_LEDC_CH    CONFIG_BSP_DISPLAY_BRIGHTNESS_LEDC_CH
+
+esp_err_t bsp_display_brightness_init(void)
+{
+    // Configure backlight GPIO
+    gpio_config_t io_conf = {
+        .pin_bit_mask = (1ULL << BSP_LCD_BACKLIGHT),
+        .mode = GPIO_MODE_OUTPUT,
+        .pull_up_en = GPIO_PULLUP_DISABLE,
+        .pull_down_en = GPIO_PULLDOWN_DISABLE,
+        .intr_type = GPIO_INTR_DISABLE,
+    };
+    return gpio_config(&io_conf);
+}
+
+esp_err_t bsp_display_brightness_set(int brightness_percent)
+{
+    if (brightness_percent > 100) {
+        brightness_percent = 100;
+    }
+    if (brightness_percent < 0) {
+        brightness_percent = 0;
+    }
+
+    ESP_LOGI(TAG, "Setting LCD backlight: %d%%", brightness_percent);
+
+    return gpio_set_level(BSP_LCD_BACKLIGHT, brightness_percent > 0);
+}
+
+esp_err_t bsp_display_backlight_off(void)
+{
+    return bsp_display_brightness_set(0);
+}
+
+esp_err_t bsp_display_backlight_on(void)
+{
+    return bsp_display_brightness_set(100);
+}
+
+esp_err_t bsp_display_new(const bsp_display_config_t *config, esp_lcd_panel_handle_t *ret_panel,
+                          esp_lcd_panel_io_handle_t *ret_io)
+{
+    esp_err_t ret = ESP_OK;
+    assert(config != NULL && config->max_transfer_sz > 0);
+
+    BSP_ERROR_CHECK_RETURN_ERR(bsp_feature_enable(BSP_FEATURE_LCD, true));
+
+    /* Initialize SPI */
+    ESP_RETURN_ON_ERROR(bsp_spi_init(config->max_transfer_sz), TAG, "");
+
+    ESP_LOGI(TAG, "Install panel IO");
+    const esp_lcd_panel_io_spi_config_t io_config = {
+        .dc_gpio_num       = BSP_LCD_DC,
+        .cs_gpio_num       = BSP_LCD_CS,
+        .pclk_hz           = BSP_LCD_PIXEL_CLOCK_HZ,
+        .lcd_cmd_bits      = LCD_CMD_BITS,
+        .lcd_param_bits    = LCD_PARAM_BITS,
+        .spi_mode          = 0,
+        .trans_queue_depth = 10,
+    };
+    ESP_GOTO_ON_ERROR(esp_lcd_new_panel_io_spi((esp_lcd_spi_bus_handle_t)BSP_LCD_SPI_NUM, &io_config, ret_io), err, TAG,
+                      "New panel IO failed");
+
+    ESP_LOGI(TAG, "Install LCD driver");
+    const esp_lcd_panel_dev_config_t panel_config = {
+        .reset_gpio_num = BSP_LCD_RST,  // Shared with Touch reset
+        .color_space    = BSP_LCD_COLOR_SPACE,
+        .bits_per_pixel = BSP_LCD_BITS_PER_PIXEL,
+    };
+
+    ESP_GOTO_ON_ERROR(esp_lcd_new_panel_ili9341(*ret_io, &panel_config, ret_panel), err, TAG, "New panel failed");
+
+    esp_lcd_panel_reset(*ret_panel);
+    esp_lcd_panel_init(*ret_panel);
+    esp_lcd_panel_invert_color(*ret_panel, true);
```

Comment:
I tried this BSP with `examples/display` example and I had to change this to `false` to get correct colors.

Does it work on your device? Is it possible that I have some older version of M5Stack Core that does not need color inversion?
```suggestion
    esp_lcd_panel_invert_color(*ret_panel, false);
```

Reply:
![image](https://github.com/user-attachments/assets/e760ef31-b2b2-4dea-bbb4-b94cc4e006df)
I think it's because of the screen change?

Reply:
We should do something here to avoid that. Do you have any good ideas?

Reply:
I tried to change it to false and it turned into a reverse color in version 2.7😂, which I think is the difference between TN screens and IPS screens
![image](https://github.com/user-attachments/assets/0a3bf902-bd88-468b-9f00-8fecec914197)
![image](https://github.com/user-attachments/assets/87c0de36-188f-4f47-bd91-a2ab9285f637)

Reply:
So it works for you correctly with `esp_lcd_panel_invert_color(*ret_panel, true);`, right?

I checked you code from here https://github.com/m5stack/M5Unified/blob/master/src/M5Unified.cpp and it does not differentiate between Core versions...

Ideally, we would identify the board version from software and the set the color inversion accordingly. If this is not possible, please add a comment there that users of older M5Stack Core must set the color inversion to `false`

Reply:
I added the inverse color related configuration in the new commit, and added the description in `Kconfig` and `README`